facebookresearch/LIGHT
# Episode compression

**Patch description**
At the moment content loggers would save _nearly_ the _entire_ room multiple times as events came through. This makes the data format for LIGHT unwieldy and massive, and as we're moving to a new setup and planning to release tasks from here, this is not great.

The key implementation is two parts: First, we create a "compressed" mode for events, which replace storing the nodes as just storing their IDs in the output JSON. This is useful when we know the starting graph and everything should be an event-driven change from that point on.

Now, `GoEvent`s are special, as they may move an agent from room to room, and `RoomContentLoggers` only have information about their current room. As such, these events need to store the entire tree of what the agent is carrying.

This PR also contains a lot of Typing cleanup.

**Testing steps**
```
>>  pytest light
..s............................................sssssssssssss.........ssssssssss..s..s..s..s......................................................................................................................... [ 96%]
.......                                                                                                                                                                                                              [100%]
===================================================================================================== warnings summary =====================================================================================================
/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/numexpr/expressions.py:21
/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/numexpr/expressions.py:21
  /Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/numexpr/expressions.py:21: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    _np_version_forbids_neg_powint = LooseVersion(numpy.__version__) >= LooseVersion('1.12.0b1')

/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/scipy/special/orthogonal.py:79
/Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/scipy/special/orthogonal.py:79
  /Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/scipy/special/orthogonal.py:79: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    from numpy import (exp, inf, pi, sqrt, floor, sin, cos, around, int,

light/world/tests/test_agent_death.py::TestInteractionLoggers::test_run
  /Users/jju/miniconda3/envs/py38/lib/python3.8/asyncio/events.py:81: RuntimeWarning: coroutine 'BattleRoyaleSoul.observe_event' was never awaited
    self._context.run(self._callback, *self._args)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================================================================= short test summary info ==================================================================================================
SKIPPED [1] light/data_model/tests/test_db.py:71: Deprecating
SKIPPED [8] light/graph/builders/tests/test_StarSpaceBuilder.py: Need to update starspace builders to use Model Pool
SKIPPED [5] light/graph/builders/tests/test_StarspaceNeighborBuilder.py: Need to update starspace builders to use Model Pool
SKIPPED [7] light/graph/builders/tests/test_user.py: Need to update starspace builders to use Model Pool
SKIPPED [3] /Users/jju/miniconda3/envs/py38/lib/python3.8/site-packages/_pytest/unittest.py:258: Skip GraphEventTests tests, it's a base class
SKIPPED [1] light/graph/tests/test_events.py:234: No valid events for <class 'light.graph.events.graph_events.SayEvent'>
SKIPPED [1] light/graph/tests/test_events.py:234: No valid events for <class 'light.graph.events.graph_events.ShoutEvent'>
SKIPPED [1] light/graph/tests/test_events.py:234: No valid events for <class 'light.graph.events.graph_events.WhisperEvent'>
SKIPPED [1] light/graph/tests/test_events.py:234: No valid events for <class 'light.graph.events.graph_events.TellEvent'>
191 passed, 28 skipped, 5 warnings in 14.21s
```
